src/tools/ceph_objectstore_tool.cc

```diff
@@ -434,7 +434,54 @@ int _action_on_all_objects_in_pg(ObjectStore *store, coll_t coll, action_on_obje
   return 0;
 }
 
-int action_on_all_objects_in_pg(ObjectStore *store, coll_t coll, action_on_object_t &action, bool debug)
+int action_on_all_objects_in_pg(ObjectStore *store, string pgidstr, action_on_object_t &action, bool debug)
+{
+  spg_t pgid;
+  // Scan collections in case this is an ec pool but no shard specified
+  unsigned scanned = 0;
+  int r = 0;
+  vector<coll_t> colls_to_check;
+  vector<coll_t> candidates;
+  r = store->list_collections(candidates);
+  if (r < 0) {
+    cerr << "Error listing collections: " << cpp_strerror(r) << std::endl;
+    return r;
+  }
+  pgid.parse(pgidstr.c_str());
+  for (vector<coll_t>::iterator i = candidates.begin();
+       i != candidates.end();
+       ++i) {
+    spg_t cand_pgid;
+    snapid_t snap;
+    if (!i->is_pg(cand_pgid, snap))
+      continue;
+    if (snap != CEPH_NOSNAP)
```

Comment:
I believe at one time snapshots use to be contained in the collection names.  That is why you can see 1.3_head as the collection name for pg 1.3 in filestore.  However, now the snapshots are handled by the objectstore itself within the collections.  I'm not sure when you'd ever see anything other than CEPH_NOSNAP (_head) in an spg_t unless an old filestore had that kind of collection still laying around.  We need to talk to Sam to confirm possibly getting rid of dead code and do some cleanup.